src/proxy/ISemaphore.ts

```diff
@@ -0,0 +1,66 @@
+import * as Promise from 'bluebird';
+import Long = require('long');
+import {DistributedObject} from '../DistributedObject';
+
+export interface ISemaphore extends DistributedObject {
+    /**
+     * Try to initialize this ISemaphore instance with the given permit count.
+     * @param permits
+     */
+    init(permits: number): Promise<boolean>;
+
+    /**
+     * Acquires the given number of permits if they are available, and returns
```

Comment:
I see this is copied from Java client's documentation. Unfortunately, there are major differences between Java and Node.js with respect to this proxy. Java client blocks the calling thread until semaphore acquire succeeds. Node.js client should fulfill the returned promise when acquire request succeeds, otherwise promise is simply not fulfilled. This client already behaves this way as far as I see but documentation is wrong.